docs/browser.md

```diff
@@ -0,0 +1,400 @@
+# Mattermost Browser Load Testing Agent
+
+## Overview
+
+The Browser Load Testing Agent is a specialized component of the Mattermost Load Test Framework that enables browser-based load testing using Playwright. This tool simulates real user interactions with the Mattermost web client, providing valuable insights into the platform's performance under various browser-based workloads.
+
+## Architecture
+
+### Core Components
+
+1. **Server (src/server.ts)**
+   - Entry point for the browser agent
+   - Configurable through environment variables
+   - Handles HTTP server setup and initialization
+
+2. **App (src/app.ts)**
+   - Fastify-based application setup
+   - Registers routes and middleware
+   - Configures logging and server options
+
+3. **Browser Manager (src/lib/browser_manager.ts)**
+   - Manages browser sessions using Playwright
+   - Handles browser lifecycle (creation, execution, cleanup)
+   - Implements session state management
+   - Supports parallel test execution
+
+4. **Test Scenarios (src/tests/)**
+   - Contains automated browser interactions
+   - Implements common user actions:
+     - Login
+     - Posting messages
+     - Channel navigation
+     - Scrolling through message history
+
+### Configuration
+
+The project uses TypeScript and is configured with:
+- `tsconfig.json` for TypeScript compilation
+- `playwright.config.ts` for Playwright test settings
+- `prettier.config.cjs` for code formatting
+- Environment variables for runtime configuration
+
```

Comment:
This is very thorough, but I'm a bit concerned it will be difficult to keep it updated, specially in these early stages of the browser agents. Thoughts?